compiler/src/main/scala/org/scalajs/core/compiler/PrepJSExports.scala

```diff
@@ -196,28 +200,44 @@ trait PrepJSExports { this: PrepJSInterop =>
       // Enforce no __ in name
       if (name.contains("__")) {
         // Get position for error message
-        val pos = if (annot.stringArg(0).isDefined)
-          annot.args.head.pos
-        else trgSym.pos
+        val pos = if (hasExplicitName) annot.args.head.pos else trgSym.pos
 
         reporter.error(pos,
             "An exported name may not contain a double underscore (`__`)")
       }
 
       // Make sure we do not override the default export of toString
-      if (!sym.isConstructor && name == "toString" && !named &&
-          sym.name != nme.toString_ && sym.tpe.params.isEmpty &&
-          !jsInterop.isJSGetter(sym)) {
+      def isIllegalToString = {
+        isMember && !isNamedExport &&
+        name == "toString" && sym.name != nme.toString_ &&
+        sym.tpe.params.isEmpty && !jsInterop.isJSGetter(sym)
+      }
+
+      if (isIllegalToString) {
         reporter.error(annot.pos, "You may not export a zero-argument " +
             "method named other than 'toString' under the name 'toString'")
       }
 
-      if (named && jsInterop.isJSProperty(sym)) {
+      // Don't allow apply without explicit name
+      if (isMember && !hasExplicitName && sym.name == nme.apply) {
+        if (isExportAll) {
+          reporter.warning(sym.pos, "Exported methods named apply are not " +
+              "exported to function appliction in JavaScript, but to a member " +
+              "named apply. Add @JSExport(\"apply\") to silence this warning.")
```

Comment:
Yes, we should also schedule that to become an error.
It should be an error at some point, because, if JavaScript ever offers a way to add callable capabilities to non-Function objects, then we can make it work (in ES 12 or something ^^)